networkx/utils/backends.py

```diff
@@ -972,20 +976,162 @@ def _convert_arguments(self, backend_name, args, kwargs):
+                    # below (but switched) to clear unnecessary items from the cache.
+                    # Use `list(cache.items())` to be thread-safe.
+                    for (ekey, nkey, gkey), val in list(cache.items()):
+                        if edge_key is False or ekey is True:
+                            pass
+                        elif (
+                            edge_key is True
+                            or ekey is False
+                            or not edge_key.issubset(ekey)
+                        ):
+                            continue
+                        if node_key is False or nkey is True:
+                            pass
+                        elif (
+                            node_key is True
+                            or nkey is False
+                            or not node_key.issubset(nkey)
+                        ):
+                            continue
+                        if graph_key and not gkey:
+                            continue
+                        warnings.warn(
+                            f"Using cached graph for {backend_name!r} backend in "
+                            f"call to {self.name}.\n\nFor the cache to be consistent "
+                            "(i.e., correct), the input graph must not have been "
+                            "manually mutated since the cached graph was created. "
+                            "Examples of manually mutating the graph data structures "
+                            "resulting in an inconsistent cache include:\n\n"
+                            "    >>> G[u][v][key] = val\n\n"
+                            "and\n\n"
+                            "    >>> for u, v, d in G.edges(data=True):\n"
+                            "    ...     d[key] = val\n\n"
+                            "Using methods such as `G.add_edge(u, v, weight=val)` "
+                            "will correctly clear the cache to keep it consistent. "
+                            "You may also use `G.__networkx_cache__.clear()` to "
+                            "manually clear the cache, or set `G.__networkx_cache__` "
+                            "to None to disable caching for G. Enable or disable "
+                            "caching via `nx.config.cache_converted_graphs` config."
+                        )
+                        return val
+
+        backend = _load_backend(backend_name)
+        rv = backend.convert_from_nx(
+            graph,
+            edge_attrs=edge_attrs,
+            node_attrs=node_attrs,
+            preserve_edge_attrs=preserve_edge_attrs,
+            preserve_node_attrs=preserve_node_attrs,
+            preserve_graph_attrs=preserve_graph_attrs,
+            name=self.name,
+            graph_name=graph_name,
+        )
+        if use_cache and nx_cache is not None:
+            # Remove old cached items that are no longer necessary since they
```

Comment:
I think I lost track of how this works; can there be multiple converted graphs cached for each backend?  I was under the impression the cache contains only the last converted graph for each backend, and the logic above was to support returning a cached graph that's potentially a superset (attribute-wise) of the graph needed. If that's not the case, I'm just wondering if this could use an unexpectedly large amount of memory (thinking of nx-cugraph instances and GPU memory in particular).